public/js/redux/chartsRedux.js

```diff
@@ -28,9 +30,15 @@ const chartsRedux = State({
         }
     },
 
-    updateComposerVsIncopy(state, composerVsInCopyData) {
+    updateComposerVsIncopy(state, { chartData, startDate, endDate }) {
+        const { composerResponse, inCopyResponse } = chartData;
+        const range = endDate.diff(startDate, 'days');
+        const composerData = composerResponse.data.length < range ?  fillMissingDates(startDate, endDate, composerResponse.data).sort(compareDates) : composerResponse.data;
```

Comment:
I think the data cleansing should happen in the API response rather than in here - we should have clean, ready to go data at this point, what do you think?